demo_example/async_echo_client.cpp

```diff
@@ -17,9 +17,38 @@
 #include <cstring>
 #include <iostream>
 #include "asio_util.hpp"
+#include <async_simple/coro/SpinLock.h>
+
+static int concurrency = 4;
+static async_simple::coro::SpinLock g_wlock, g_rlock;
 
 using asio::ip::tcp;
 
+async_simple::coro::Lazy<int> task(asio::ip::tcp::socket& socket) {
+    const int max_length = 1024;
+    char write_buf[max_length] = {"hello async_simple"};
+    char read_buf[max_length];
+    auto send = [&]() -> async_simple::coro::Lazy<> {
+        auto scopeLock = co_await g_wlock.coScopedLock();
```

Comment:
coScopedLock need a executor to reschedule current coroutine to avoid the others be blocked, but this demo code use syncAwait to start a lazy task (without executor), so segment fault occurred

you can use ScopedSpinLock to avoid the coroutine yield or pass a executor to syncAwait